lib/go/Makefile

```diff
@@ -15,15 +15,37 @@ endif
 export GOPATH
 
 
+################################################################################
+##                                   DEP                                      ##
+################################################################################
+
+DEP ?= ./dep
+DEP_VER ?= 0.3.1
+DEP_BIN := dep-$$GOHOSTOS-$$GOHOSTARCH
+DEP_URL := https://github.com/golang/dep/releases/download/v$(DEP_VER)/$$DEP_BIN
+
+$(DEP):
+	GOVERSION=$$(go version | awk '{print $$4}') && \
+	GOHOSTOS=$$(echo $$GOVERSION | awk -F/ '{print $$1}') && \
+	GOHOSTARCH=$$(echo $$GOVERSION | awk -F/ '{print $$2}') && \
+	DEP_BIN="$(DEP_BIN)" && \
+	DEP_URL="$(DEP_URL)" && \
+	curl -sSLO $$DEP_URL && \
+	chmod 0755 "$$DEP_BIN" && \
+	mv -f "$$DEP_BIN" "$@"
+ifneq (./dep,$(DEP))
+dep: $(DEP)
+endif
+
+vendor: | $(DEP)
+	$(DEP) ensure -v -vendor-only
+
+
 ########################################################################
 ##                             PROTOC                                 ##
 ########################################################################
 
-# Only set PROTOC_VER if it has an empty value.
```

Comment:
Hi @jieyu ,

I've always err'd on the side of caution and opted for the use of `ifeq` to determine if a variable is already set instead of `?=`. I've always assumed my method is more performant since `?=` is subject to recursive expansion. I made this change because it occurs to me if the value contains no variables then `?=` probably doesn't incur the penalty of the would-be recursive expansion and is simpler and as effective as the previous method.

Do you know if this is true? Is `?=` without variables in the value still subject to a check for recursive expansion everything the variable is referenced?

Reply:
Hi @gpaul,

Do you have any advice or guidance related to the above question?

Reply:
gnu make will short-circuit the recursive expand if it doesn't find a '$':
http://git.savannah.gnu.org/cgit/make.git/tree/expand.c#n230

Reply:
Cool, thanks!

Reply:
But seriously @gpaul, thank you for tracking this down. I know there are hundreds of locations where I've used the other pattern to ensure that things are more performant. Knowing this information is truly helpful.